compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -1062,7 +1062,14 @@ OMR::Power::TreeEvaluator::m2lEvaluator(TR::Node *node, TR::CodeGenerator *cg)
 TR::Register*
 OMR::Power::TreeEvaluator::m2vEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
-   return TR::TreeEvaluator::unImpOpEvaluator(node, cg);
+   // Resulting vector should represent boolean array
+   // The mask is already in this form, with 'true' represented as all 1's
+   TR_ASSERT_FATAL_WITH_NODE(node, node->getDataType().getVectorLength() == TR::VectorLength128 &&
+                             node->getDataType().getVectorElementType() == TR::Int8,
+                   "Only 128-bit vectors are supported %s", node->getDataType().toString());
+
+   return TR::TreeEvaluator::passThroughEvaluator(node, cg);
```

Comment:
This might work, but I don't think its semantically correct. I think the upper 7-bits should be zero.

Reply:
I think boolean` true `can be represented by -1 in Java.

Reply:
I will defer to @hzongaro on this. Must a boolean value be either 0 or 1?

Reply:
I think you are right, at least for Java. JVM Specification:
"The Java Virtual Machine encodes boolean array components using 1 to represent
true and 0 to represent false."

Reply:
Even if other languages have simply non-zero requirement I think it's better to make it 1.

Reply:
> Must a boolean value be either 0 or 1?

Just to confirm, what @gita-omr found from the JVM specification is correct - boolean values in Java must be zero or one.